server/user.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Expire time of cookie.
+ * @const {number}
+ */
+user.cookieExpireTime = 24 * 60 * 60 * 1000;
+
+/**
+ * @typedef {{
+ *   email: (string|undefined),
+ *   username: string,
+ *   password: string,
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.Info;
+
+/**
+ * @typedef {{
+ *   error: string
+ * }}
+ */
+user.Error;
+
+/**
+ * Checks the user information for signing in.
+ * @param {!mongodb.Db} db Session database.
+ * @param {string} userPath Directory of user information file.
+ * @param {!user.Info} userInfo User Information
+ * @param {function(!Object)} callback Callback function.
+ */
+user.signUp = function(db, userPath, userInfo, callback) {
+  var checkDuplicate = {
+    duplicated: false,
+    elements: []
+  };
+  var cursor = db.collection('userInfo').find({
+    $or: [{username: userInfo.username}, {email: userInfo.email}]});
+  var data = [];
+  cursor.each(function(err, doc) {
+    assert.equal(err, null, 'error occurs');
+    if (doc != null) {
+      data.push(doc);
+    } else {
+      var result = authenticateCallback();
+      callback(result);
+    }
+  });
+  var authenticateCallback = function() {
```

Comment:
I think it is better to define the authenticateCallback in the sign functions directly, because:
1. Different sign functions have different authenticateCallback functions and authenticateCallback functions are not generic.
2. The authenticateCallback will not be used in any other places besides the sign functions.